multiversx_sdk_cli/native_auth_client.py

```diff
@@ -0,0 +1,104 @@
+import base64
+import json
+from typing import Any, Dict, Optional
+
+import requests
+
+from multiversx_sdk_cli.errors import NativeAuthClientError
+
+EXPIRY_TIME_IN_SECONDS = 60 * 60 * 2
+DEFAULT_API_URL = "https://api.multiversx.com"
+
+
+class NativeAuthClientConfig:
+    def __init__(
+            self,
+            origin: str = '',
+            api_url: str = DEFAULT_API_URL,
+            expiry_seconds: int = EXPIRY_TIME_IN_SECONDS,
+            block_hash_shard: Optional[int] = None,
+            gateway_url: Optional[str] = None,
+            extra_request_headers: Optional[Dict[str, str]] = None
+    ) -> None:
+        self.origin = origin
+        self.api_url = api_url
+        self.expiry_seconds = expiry_seconds
+        self.block_hash_shard = block_hash_shard
+        self.gateway_url = gateway_url
+        self.extra_request_headers = extra_request_headers
+
+
+class NativeAuthClient:
+    def __init__(self, config: Optional[NativeAuthClientConfig] = None) -> None:
+        self.config = config or NativeAuthClientConfig()
+
+    def get_token(self, address: str, token: str, signature: str) -> str:
```

Comment:
Functions can be re-ordered to match the usual flow. E.g. first "initialize", then ...